### PR TITLE
build: add code coverage 

### PR DESCRIPTION
**Pull request by CJ42**

### CODE REVIEW

File: .github/workflows/build-lint-test.yml [hunk 1]
```diff
@@ -87,3 +87,12 @@ jobs:
 
       - name: run tests
         run: npm run test:${{ matrix.lsp }}
+
+      # Trigger Coverage
+      - name: Trigger Code Coverage Workflow
+        if: github.event_name == 'push'
```
Comment (CallumGrindle) [inline .github/workflows/build-lint-test.yml:93]: I think we don't need this conditional, this workflow will only run on pull requests so this wont ever get triggered

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      # This will also generate the Typechain types used by the Chai tests
+      - name: Install dependencies
+        if: steps.build-cache.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - name: Run tests with coverage
+        run: npm run test:coverage
+
+      - name: Upload to Coveralls
```
Comment (CallumGrindle) [inline .github/workflows/coverage.yml:39]: lgtm, lets see how it works when testing it

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
 
-name: Test + Generate coverage report
+name: Solidity Code coverage
 
 on:
-  workflow_dispatch:
-    types: [trigger-coverage]
+  push:
+    branches:
+      - "develop"
```
Comment (Hugoo) [inline .github/workflows/coverage.yml:10]: can you use checkout v3 ?

**Review by Hugoo — APPROVED** [review]

File: .github/workflows/coverage.yml (hunk 3 above)
Comment (CJ42) [reply .github/workflows/coverage.yml:10]: done

**Review by Hugoo — APPROVED** [review]
nice!

**Pull request merged**